### PR TITLE
[ui, job status] Make panel status mode a queryParam

### LINKED ISSUE #16340: Service Job Panel: Make current/historical a query parameter
Make it so the URL is affected by and affects these:
image

### PR DESCRIPTION
**Pull request by philrenaud**
Resolves #16340 

Passes `statusMode` down to the job status panel; changes behaviour to be at job index level.

### CODE REVIEW

File: ui/app/controllers/jobs/job/index.js [hunk 1]
```diff
@@ -31,6 +33,11 @@ export default class IndexController extends Controller.extend(
   sortDescending = false;
   activeTask = null;
 
+  /**
+   * @type {('current'|'historical')}
+   */
+  statusMode = 'current';
```
Comment (ChaiWithJai) [inline ui/app/controllers/jobs/job/index.js:39]: `question`: Shouldn't this be a `tracked` property? I think despite being a `classic` controller `tracked` should be [backwards compatible](https://guides.emberjs.com/release/upgrading/current-edition/tracked-properties/#toc_backwards-compatibility). We do this in the `Tokens` controller I believe.

File: ui/app/controllers/jobs/job/index.js [hunk 2]
```diff
@@ -39,4 +46,12 @@ export default class IndexController extends Controller.extend(
       this.set('activeTask', null);
     }
   }
+
+  /**
+   * @param {('current'|'historical')} mode
+   */
+  @action
+  setStatusMode(mode) {
+    this.set('statusMode', mode);
```
Comment (ChaiWithJai) [inline ui/app/controllers/jobs/job/index.js:55]: Once you edit to `tracked` it should be:

```
this.statusMode = mode;
```

File: ui/app/controllers/jobs/job/index.js (hunk 1 above)
Comment (ChaiWithJai) [reply ui/app/controllers/jobs/job/index.js:39]: `question`: Are you going to want to fetch data when this query parameter changes? If so, you'll need to update the associated `Route`?

File: ui/app/components/job-status/panel.hbs [hunk 3]
```diff
@@ -1,27 +1,27 @@
...
       </button>
     </div>
   </div>
   <div class="boxed-section-body">
-    {{#if (eq this.mode "current")}}
+    {{#if (eq @statusMode "current")}}
       <h3 class="title is-4 running-allocs-title"><strong>{{@job.runningAllocs}}/{{this.totalAllocs}}</strong> Allocations Running</h3>
 
-      <JobStatus::AllocationStatusRow @allocBlocks={{this.allocBlocks}} />
+      aaa<JobStatus::AllocationStatusRow @allocBlocks={{this.allocBlocks}} />bbb
```
Comment (ChaiWithJai) [inline ui/app/components/job-status/panel.hbs:24]: Looks like we don't want `aaa` and `bbb` in prod.

File: ui/app/controllers/jobs/job/index.js (hunk 1 above)
Comment (ChaiWithJai) [inline ui/app/controllers/jobs/job/index.js:39]: `question`: Is this better off as a boolean?

**Review by ChaiWithJai — CHANGES_REQUESTED** [review]
Minor changes required.

File: ui/app/controllers/jobs/job/index.js (hunk 1 above)
Comment (philrenaud) [reply ui/app/controllers/jobs/job/index.js:39]: I can try to make this tracked, but we don't typically do this in the Nomad UI I've found so I've got some concerns with bringing glimmer concepts in.

Comment (philrenaud) [reply ui/app/controllers/jobs/job/index.js:39]: I don't think so; we don't save anything on perf and keeping it a specific string lets us extend this when we want to.

Comment (philrenaud) [reply ui/app/controllers/jobs/job/index.js:39]: Also just from a legibility standpoint, `statusMode=historical` reads cleaner than `statusModeHistorical=true` for an end-user.

**Review by ChaiWithJai — APPROVED** [review]
My only critique here is that we're not dispatching a new network request upon updating this query parameter by calling a refresh in the associated route. We're relying on derived state computations for `allocations` and `taskGroups` for this. Thankfully, we do have watchers which **should** update upon the server detecting a change. However, as we've noted, some browsers can only keep 2 concurrent requests open and `jobs.job.index` is watching about half a dozen entities so we might be breaking on real-time updates and not realize that.

**Pull request merged**